homeassistant/components/overkiz/climate_entities/atlantic_pass_apc_heating_and_cooling_zone.py

```diff
@@ -0,0 +1,134 @@
+    OverkizCommandParam.ECO: PRESET_ECO,
+    OverkizCommandParam.INTERNAL_SCHEDULING: PRESET_COMFORT,
+}
+
+PRESET_MODES_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_PRESET_MODES.items()}
+
+THERMAL_CONFIGURATION_STATE = "core:ThermalConfigurationState"
+
+
+class AtlanticPassAPCHeatingAndCoolingZone(OverkizEntity, ClimateEntity):
+    """Representation of Atlantic Pass APC Heating And Cooling Zone Zone Control."""
+
+    _attr_hvac_modes = [*HVAC_MODE_TO_OVERKIZ]
+    _attr_preset_modes = [*PRESET_MODES_TO_OVERKIZ]
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.PRESET_MODE
+    )
+    _attr_temperature_unit = TEMP_CELSIUS
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        return OVERKIZ_TO_HVAC_MODE[
+            cast(str, self.executor.select_state(OverkizState.IO_PASS_APC_HEATING_MODE))
+        ]
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set new target hvac mode."""
+        commands = [
+            Command(
+                OverkizCommand.SET_PASS_APC_HEATING_MODE,
+                [HVAC_MODE_TO_OVERKIZ[hvac_mode]],
+            ),
+            # We also needs to execute these 3 commands to make it work correctly
+            Command(
+                OverkizCommand.SET_DEROGATION_ON_OFF_STATE,
+                [OverkizCommandParam.OFF],
+            ),
+            Command(OverkizCommand.REFRESH_PASS_APC_HEATING_MODE),
+            Command(OverkizCommand.REFRESH_PASS_APC_HEATING_PROFILE),
+        ]
+        await self.executor.async_execute_commands(commands)
+
+    @property
+    def preset_mode(self) -> str:
+        """Return the current preset mode, e.g., home, away, temp."""
+        return OVERKIZ_TO_PRESET_MODES[
+            cast(str, self.executor.select_state(OverkizState.IO_PASS_APC_HEATING_MODE))
+        ]
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+        commands = [
+            Command(
+                OverkizCommand.SET_PASS_APC_HEATING_MODE,
+                [PRESET_MODES_TO_OVERKIZ[preset_mode]],
+            ),
+            # We also needs to execute these 3 commands to make it work correctly
+            Command(
+                OverkizCommand.SET_DEROGATION_ON_OFF_STATE,
+                [OverkizCommandParam.OFF],
+            ),
+            Command(OverkizCommand.REFRESH_PASS_APC_HEATING_MODE),
+            Command(OverkizCommand.REFRESH_PASS_APC_HEATING_PROFILE),
+        ]
+
+        await self.executor.async_execute_commands(commands)
+
+    @property
+    def hvac_action(self) -> str:
+        """Return hvac current action."""
+        return OVERKIZ_TO_HVAC_MODE[
+            cast(str, self.executor.select_state(THERMAL_CONFIGURATION_STATE))
+        ]
+
+    @property
+    def target_temperature(self) -> float:
+        """Return hvac target temperature."""
+        return cast(
```

Comment:
Can you please check if you can use [value_as_float](https://github.com/iMicknl/python-overkiz-api/blob/main/pyoverkiz/models.py#L304) to avoid the cast. 
`self.executor.device.states[OverkizState.CORE_TARGET_TEMPERATURE].value_as_float():
`

Reply:
While doing that, I'm getting the error:
`AttributeError: 'float' object has no attribute 'value_as_float'` 
I'll let as it is now

Reply:
Can you please show your code and the full stack error?

Reply:
With the current code, here is the error if I remove all the `cast(float` :  

```
homeassistant/components/overkiz/climate_entities/atlantic_pass_apc_heating_and_cooling_zone.py:80: error: Incompatible return value type (got "Union[str, float, Dict[str, Any], List[Any], None]", expected "Optional[float]")  [return-value]
homeassistant/components/overkiz/climate_entities/atlantic_pass_apc_heating_and_cooling_zone.py:155: error: Incompatible return value type (got "Union[str, float, Dict[str, Any], List[Any], None]", expected "float")  [return-value]
homeassistant/components/overkiz/climate_entities/atlantic_pass_apc_heating_and_cooling_zone.py:159: error: Incompatible return value type (got "Union[str, float, Dict[str, Any], List[Any], None]", expected "float")  [return-value]
homeassistant/components/overkiz/climate_entities/atlantic_pass_apc_heating_and_cooling_zone.py:163: error: Incompatible return value type (got "Union[str, float, Dict[str, Any], List[Any], None]", expected "float")  [return-value]
homeassistant/components/overkiz/climate_entities/atlantic_pass_apc_heating_and_cooling_zone.py:167: error: Incompatible return value type (got "Union[str, float, Dict[str, Any], List[Any], None]", expected "float")  [return-value]
```